sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1487,29 +1487,44 @@ class GpuRunningWindowIterator(
     val fixers = fixerIndexMap
     val numRows = cb.numRows()
 
+    var newOrder: Option[Array[Scalar]] = None
+
     withResource(computeBasicWindow(cb)) { basic =>
       withResource(GpuProjectExec.project(cb, boundPartitionSpec)) { parts =>
         val partColumns = GpuColumnVector.extractBases(parts)
         withResourceIfAllowed(arePartsEqual(lastParts, partColumns)) { partsEqual =>
-          val fixedUp = if (fixerNeedsOrderMask) {
-            withResource(GpuProjectExec.project(cb, boundOrderColumns)) { order =>
-              val orderColumns = GpuColumnVector.extractBases(order)
-              // We need to fix up the rows that are part of the same batch as the end of the
-              // last batch
-              withResourceIfAllowed(areOrdersEqual(lastOrder, orderColumns, partsEqual)) {
-                orderEqual =>
-                  closeOnExcept(fixUpAll(basic, fixers, partsEqual, Some(orderEqual))) { fixed =>
-                    saveLastOrder(getScalarRow(numRows - 1, orderColumns))
-                    fixed
+
+          // we backup the fixers state and restore it in the event of a retry
+          val fixedUp = withRetryNoSplit {
```

Comment:
In order for the retry to work properly we need all of the data on the GPU to be spillable at that point in time. Here cb is not spillable so even though we pause the calculation and roll back we do not go back to a point when we could release everything from the GPU.